### PR TITLE
Update NativeTests.exe to use TestHooks, add UnicodeText tests from #4905

### PR DESCRIPTION
**Pull request by jackhorton**
@aileronajay did the original work for this PR, but adding TestHooks to NativeTests.exe was required and a bit too involved for a first-time contributor. Hopefully this will allow more direct native unit testing in the future.

### CODE REVIEW

**Comment by dilijev** [comment]
@jackhorton nit: ideally the original commit will be in the history, or the author(original)/committer(you) of the commit we merge here should be set so that the original author gets credit in our history.

File: bin/NativeTests/UnicodeTextTests.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        Test(_u("1"), _u("2"), -1);
+        Test(_u("2"), _u("1"), 1);
+        Test(_u("10"), _u("01"), 1);
+        Test(_u("01"), _u("10"), -1);
+        Test(_u("01"), _u("1"), -1);
+        Test(_u("1"), _u("01"), 1);
+    }
+
+
```
Comment (dilijev) [inline bin/NativeTests/UnicodeTextTests.cpp]: nit: extra newline

**Comment by jackhorton** [comment]
At this point its much easier for me to just --amend @aileronajay as the author than to try to rebuild the history.

**Comment by dilijev** [comment]
@jackhorton that's perfect

**Pull request merged**